File: src/main/java/it/unibo/crabinv/model/entities/enemies/rewardservice/EnemyRewardService.java
package it.unibo.crabinv.model.entities.enemies.rewardservice;

import it.unibo.crabinv.model.entities.enemies.Enemy;
import it.unibo.crabinv.model.save.GameSession;

/**
 * It's the implementation of the reward service.
 */
public final class EnemyRewardService implements RewardsService {
    private final GameSession profile;

    /**
     * It's the constructor of the reward service.
     *
     * @param profile it's the profile needed to give the reward
     */
    public EnemyRewardService(final GameSession profile) {
        this.profile = profile;
    }

    /**
     * {@inheritDoc}
     *
     * @param enemy the dead enemy
     */
    @Override
    public void rewardEnemyDeath(final Enemy enemy) {
        profile.addCurrency(enemy.getReward());
    }
}

File: src/main/java/it/unibo/crabinv/model/entities/enemies/rewardservice/EnemyRewardService.kt
package it.unibo.crabinv.model.entities.enemies.rewardservice

import it.unibo.crabinv.model.core.save.GameSession
import it.unibo.crabinv.model.entities.enemies.Enemy

/**
 * It's the implementation of the reward service.
 * @param profile it's the profile needed to give the reward
 */
class EnemyRewardService(
    private val profile: GameSession,
) : RewardsService {
    override fun rewardEnemyDeath(enemy: Enemy) {
        profile.addCurrency(enemy.reward)
    }
}
